src/lib.rs

```diff
@@ -59,19 +59,8 @@
     unused_results,
     warnings
 )]
-#![cfg_attr(
-    any(
-        target_os = "redox",
-        all(
-            not(test),
-            not(feature = "use_heap"),
-            unix,
-            not(any(target_os = "macos", target_os = "ios")),
-            any(not(target_os = "linux"), feature = "dev_urandom_fallback")
-        )
-    ),
-    no_std
-)]
+// std is only used in tests or when gated with #[cfg(use_heap)]
+#![cfg_attr(not(any(test, feature = "use_heap")), no_std)]
```

Comment:
The comment here isn't necessary since it's just duplicating the code, so let's remove it.